core/src/main/java/io/confluent/rest/RestConfig.java

```diff
@@ -1139,6 +1168,72 @@ public final Map<String, String> getMap(String propertyName) {
     return map;
   }
 
+  static List<NamedURI> parseListeners(
+      List<String> listeners,
+      Map<String,String> listenerProtocolMap,
+      int deprecatedPort,
+      List<String> supportedSchemes,
+      String defaultScheme) {
+
+    // handle deprecated case, using PORT_CONFIG.
+    // TODO: remove this when `PORT_CONFIG` is deprecated, because LISTENER_CONFIG
+    // will have a default value which includes the default port.
+    if (listeners.isEmpty() || listeners.get(0).isEmpty()) {
+      listeners = singletonList(defaultScheme + "://0.0.0.0:" + deprecatedPort);
+    }
+
+    List<NamedURI> uris = listeners.stream()
+        .map(listener -> constructNamedURI(listener, listenerProtocolMap, supportedSchemes))
+        .collect(Collectors.toList());
+    List<NamedURI> namedUris =
+        uris.stream().filter(uri -> uri.getName() != null).collect(Collectors.toList());
+    List<NamedURI> unnamedUris =
+        uris.stream().filter(uri -> uri.getName() == null).collect(Collectors.toList());
+
+    if (namedUris.stream().map(a -> a.getName()).distinct().count() != namedUris.size()) {
+      throw new ConfigException(
+          "More than one listener was specified with same name. Listener names must be unique.");
+    }
+    if (namedUris.isEmpty() && unnamedUris.isEmpty()) {
```

Comment:
this should be same as uris.isEmpty.()
ok to keep this as is since this is code refactoring.

Reply:
Done.